pkg/controller/common/errors.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NonRetriableCtrlError wraps errors with the addition of having
+// the information for the error being non-retriable
+type NonRetriableCtrlError struct {
+	err      error
+	canRetry bool
+}
+
+func (cerr NonRetriableCtrlError) Error() string {
+	return cerr.err.Error()
+}
+
+// blank assignment to verify that RetriableCtrlError implements error
+var _ error = &NonRetriableCtrlError{}
+
+// IsRetriable exposes if the error is retriable or not
+func (cerr NonRetriableCtrlError) IsRetriable() bool {
+	return cerr.canRetry
+}
+
+// WrapNonRetriableCtrlError wraps an error with the RetriableCtrlError interface
+func WrapNonRetriableCtrlError(err error) *NonRetriableCtrlError {
+	return &NonRetriableCtrlError{
+		canRetry: false,
+		err:      err,
+	}
+}
+
+// IsRetriable returns whether the error is retriable or not using the
+// NonRetriableCtrlError interface
+func IsRetriable(err error) bool {
+	ccErr, ok := err.(*NonRetriableCtrlError)
```

Comment:
Do you know if we could switch to go 1.13 in the project? This would allow us to simplify the code he a little. But it's really not a big deal and probably should be done in a follow-up patch (to also wrap errors propagating up with %w).

Reply:
I tried but I didn't figure out what image to use. Go-toolset from red hat is still using 1.12

Reply:
OK